bbc/simorgh
# WSTEAM1-476 - Show Russian email promotion banner on Optimo articles (V1)

Resolves JIRA [WSTEAM1-476](https://jira.dev.bbc.co.uk/browse/WSTEAM1-476)

Overall changes
======
- Show Russian email promotion banner on Optimo articles, in addition to the CPSAssets where it is already being shown.
- Update `disclaimer` component following component standards and new UX/ A11y requirments. 

- Note V1 (or dirty version) refers to the fact we are using toggles in Simorgh. The ultimate goal is to use toggles in the BFF so that we can insert the disclaimer into the BFF response.

Example on Optimo page:

![Screenshot 2023-07-04 at 16 18 32](https://github.com/bbc/simorgh/assets/98817636/ee50f12a-678c-4cff-86c8-f6cd1a4c0c74)

Code changes
======
**data/...**
- Update fixture data for articles & cpsAssets so that they mirror the BFF response and can be rendered locally for testing.
- Add a new article for the Russian service.

**src/app/components**
- Update Disclaimer component based on component standards.
- Update Unit tests. (See Questions/ To Do)
- Update storybook file, create README and add metadata.

**src/app/pages/...**
- Update imports into necessary CPSAsset pages to use new disclaimer block
- Update Article Page to render disclaimer with new disclaimer block.

**src/app/routes/...**
- Update augmentWithDiclaimer function so that it works with both CPS and Optimo assets.
- Add position argument to augmentWithDiclaimer to decide whether it renders before the timestamp (Optimo) or after the timestamp (CPS)
- Update `article/getInitialData/index.ts` so that it renders the disclaimer on Optimo pages based on Toggles (this is a working version until we can use toggles in the BFF).

Questions/ Notes for Code Review
======
- Disclaimer unit tests - I feel like these should be updated to be more inline with new standards.
- Disclaimer story - These uses service config data as I'm unsure how to mock this.
- augmentWithTImestamp unit tests - I feel these should these be updated to reflect Optimo pages
- routes/article/getInitialData unit tests - should tests be added here to check the disclaimer is added?

Testing
======
1. Check out this PR `WSTEAM1-476`
2. Launch the development server with the command `yarn dev`
3. Visit the following links to check the behaviour matches the ACs on the ticket

(Should Render)
- Russian Article Page: http://localhost:7080/russian/articles/cjejv43ylr0o
- Russian Article Page (another example): http://localhost:7080/russian/articles/ck7pz7re3zgo
- Russian Article .amp Page: http://localhost:7080/russian/articles/cjejv43ylr0o.amp
- Russian Article .app Page: http://localhost:7080/russian/articles/cjejv43ylr0o.app
- Russian STY Page: http://localhost:7080/russian/news-55041160
- Russian STY .amp Page: http://localhost:7080/russian/features-54391793.amp
- Russian PGL Page: http://localhost:7080/russian/features-45782775
- Russian MAP Page: http://localhost:7080/russian/media-49281069

(Should not render)
- Non-Russian Article Page: http://localhost:7080/afrique/articles/cz216x22106o
- Non-Russian STY Page: http://localhost:7080/kyrgyz/23292889
- Non-Russian PGL Page: http://localhost:7080/azeri/azerbaijan-44208474
- Non-Russian MAP Page: http://localhost:7080/pidgin/23248703

5. View the component on Storybook - https://wsteam1-476--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/new-components-disclaimer--disclaimer

An A11y swarm will be required for this ticket.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._
- Designs inc Screen Reader UX: https://www.figma.com/file/U5b55Sl34Vh6vDxRKMwev1/Russian-service-disclaimer-Optimo-PW-FINAL-20230622?type=design&node-id=113%3A379&mode=design&t=aliHubWDAfsBKFpa-1
- A11y ACs: https://paper.dropbox.com/doc/Information-panel-formerly-Russian-Disclaimer-accessibility-acceptance-criteria--B7aaG0aTtvJgbudZi_JkAbFEAg-kR9WZaIuJi73YyUIl3XKw

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
